task/03-date-tasks.js

```diff
@@ -76,7 +76,29 @@ function isLeapYear(date) {
  *    Date(2000,1,1,10,0,0),  Date(2000,1,1,15,20,10,453)   => "05:20:10.453"
  */
 function timeSpanToString(startDate, endDate) {
-   throw new Error('Not implemented');
+   const timespanInMs = endDate.getTime() - startDate.getTime();
+   let milliseconds = Math.abs(timespanInMs);
+   const days = Math.floor(milliseconds / (1000 * 60 * 60 * 24));
+
+   milliseconds -= days * (1000 * 60 * 60 * 24);
+
+   let hours = Math.floor(milliseconds / (1000 * 60 * 60));
+   milliseconds -= hours * (1000 * 60 * 60);
+
+   const mins = Math.floor(milliseconds / (1000 * 60));
+   milliseconds -= mins * (1000 * 60);
+
+   const seconds = Math.floor(milliseconds / (1000));
+   milliseconds -= seconds * (1000);
+   const formatToDoubleDigit = (number) => number < 10 ? `0${number}` : `${number}`;
+   const formatToThreeDigit = (number) => number < 10
+      ? `00${number}`
+      : number > 10 && number < 100
+         ? `0${number}`
+         : number;
```

Comment:
Please,avoid this code style, if-else is more readable than multiple conditional operators ? :